cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1945,9 +1949,28 @@ func armParameterFileValue(paramType ParameterType, value any) any {
 				return intVal
 			}
 		}
+	case ParameterTypeString:
+		// Use Cases
+		// 1. Non-empty input value, return input value (no prompt)
+		// 2. Empty input value and no default - return nil (prompt user)
+		// 3. Empty input value and non-empty default - return empty input string (no prompt)
+		paramVal, paramValid := value.(string)
+		defaultVal, hasDefault := defaultValue.(string)
+		if hasDefault {
+			if paramValid && paramVal != defaultVal {
+				return paramVal
+			}
+		}
+
+		if paramValid && paramVal != "" {
+			return paramVal
+		}
+	// All other parameter types return the specified value as-is
+	default:
+		return value
 	}
 
-	return value
+	return nil
```

Comment:
Returning `nil` here regresses when the parameter is already a `bool` or `number`, the value in parameters.json doesn't get used:

![image](https://github.com/Azure/azure-dev/assets/2322434/d520475a-01d7-432b-bb1c-34c887260228)

In the old version this worked fine.

Reply:
I will take a look at this.

Reply:
@weikanglim - made a couple updates to address this.